//python -m grpc_tools.protoc -I . --python_out=. --grpc_python_out=. .\core\rpc\type.proto

syntax = "proto3";

package hello;


service Channel {
    rpc Option(OptionCode) returns(ResponseCode) {}
    rpc FrameToFrame (Request) returns (Response) {}
    rpc FrameToStream (Request) returns (stream Response) {}
    rpc StreamToFrame (stream Request) returns (Response) {}
    rpc StreamToStream (stream Request) returns (stream Response) {}
}

message OptionCode {uint32 code = 1;}
message ResponseCode {uint32 code = 1;}

message Request {
    string package = 1;
    string handler = 2;

    Event event = 3;

    string message = 4;
    repeated string messageList = 5;
    map<string, string> messageDict = 6;

    bytes image = 7;
    repeated  bytes imageList = 8;
    map<string, bytes> imageDict = 9;

    Other other = 10;
    repeated Other otherList = 11;
    map<string, Other> otherDict = 12;
}

message Response {
    string message = 1;
    repeated string messageList = 2;
    map<string, string> messageDict = 3;

    bytes image = 4;
    repeated  bytes imageList = 5;
    map<string, bytes> imageDict = 6;

    Other other = 7;
    repeated Other otherList = 8;
    map<string, Other> otherDict = 9;
}

message Event {
    string msg = 1;
    User sender = 2;
    Group group = 3;
    repeated User atList = 4;
    repeated bytes imageList = 5;
}

message User {
    string qq = 1;
    string name = 2;
}

message Group {
    string qq = 1;
    string name = 2;
}

message Other {
    bytes binary = 1;
    map<string, string> context = 2;
}